cwms-data-api/src/main/java/cwms/cda/data/dto/Pool.java

```diff
@@ -1,246 +1,249 @@
-			builder.withPoolName(poolName);
-			builder.withProjectId(matcher.group(2));
-			builder.withBottomLevelId(null);
-			builder.withTopLevelId(null);
-			builder.withImplicit(true);
-			retval = builder.build();
-		}
-
-		return retval;
-	}
-
-	// This is used in the Pools cursor.
-	@Override
-	public String toString(){
-		StringBuilder builder = new StringBuilder();
-		PoolNameType poolName = getPoolName();
-
-		builder.append(poolName.getOfficeId())
-				.append("/").append(getProjectId())
-				.append(":").append(poolName.getPoolName());
-		return builder.toString();
-	}
-
-	public static class Builder
-	{
-		private PoolNameType poolName;
-
-		private String projectId;
-		private String bottomLevelId;
-		private String topLevelId;
-		private boolean isImplicit;
-		private Number attribute;
-		private String description;
-		private String clobText;
-
-
-		public PoolNameType getPoolName()
-		{
-			return poolName;
-		}
-
-		public Builder withPoolName(PoolNameType poolName)
-		{
-			this.poolName = poolName;
-			return this;
-		}
-
-		public String getProjectId()
-		{
-			return projectId;
-		}
-
-		public Builder withProjectId(String projectId)
-		{
-			this.projectId = projectId;
-			return this;
-		}
-
-		public String getBottomLevelId()
-		{
-			return bottomLevelId;
-		}
-
-		public Builder withBottomLevelId(String bottomLevelId)
-		{
-			this.bottomLevelId = bottomLevelId;
-			return this;
-
-		}
-
-		public String getTopLevelId()
-		{
-			return topLevelId;
-		}
-
-		public Builder withTopLevelId(String topLevelId)
-		{
-			this.topLevelId = topLevelId;
-			return this;
-		}
-
-		public boolean isImplicit()
-		{
-			return isImplicit;
-		}
-
-		public Builder withImplicit(boolean implicit)
-		{
-			isImplicit = implicit;
-			return this;
-		}
-
-		public Number getAttribute()
-		{
-			return attribute;
-		}
-
-		public Builder withAttribute(Number attribute)
-		{
-			this.attribute = attribute;
-			return this;
-		}
-
-		public String getDescription()
-		{
-			return description;
-		}
-
-		public Builder withDescription(String description)
-		{
-			this.description = description;
-			return this;
-		}
-
-		public String getClobText()
-		{
-			return clobText;
-		}
-
-		public Builder withClobText(String clobText)
-		{
-			this.clobText = clobText;
-			return this;
-		}
-
-		public static Builder newInstance()
-		{
-			return new Builder();
-		}
-
-		public Pool build()
-		{
-			return new Pool(this);
-		}
-
-		public Builder withPoolType(PoolType poolType)
-		{
-			withPoolName(poolType.getPoolName());
-			withBottomLevelId(poolType.getBottomLevelId());
-			withTopLevelId(poolType.getTopLevelId());
-			withProjectId(poolType.getProjectId());
-			withImplicit(poolType.isImplicit());
-
-			return this;
-		}
-	}
-
-	@Override
-	public void validate() throws FieldException {
-		// TODO Auto-generated method stub
-
-	}
-
-
+public class Pool extends CwmsDTOBase {
+
+    private final PoolNameType poolName;
```

Comment:
Recommend this dependency be changed. No longer use PoolNameType and instead add its attributes to Pool. If not possible, then introduce a new PoolId class that has the office and pool name variables.

Reply:
Added as an item to accomplish in: https://github.com/USACE/cwms-data-api/issues/786